CONTRIBUTING.md

```diff
@@ -0,0 +1,172 @@
+
+We also always high appreciate contributions to milestones and we love to work closely with the community.
+So if you'd like to work on an issue that is scheduled for a release, please make sure to stay in close contact with the
+maintainers and make sure that it's possible for you to propose a [pull request](#pull-requests) before the planned
+release date of the milestone, if specified.  
+
+#### Unsure where to begin contributing to Citrus?
+You can start by looking for issues labeled with `good first issue` which should only require a few lines of code, and
+a test or two. 
+
+### Pull Requests
+Pull requests are required for every contribution to Citrus including contributions from maintainers and administrators.
+Before you propose a pull request, please ensure, that the [definition of done](#definition-of-done) is fulfilled.
+Please ensure that every pull request is linked to an issue. If you open a pull request without an existing issue, please 
+open an issue as well, to point out the requirements for the change. That helps us to separate the functional discussion 
+(issue) from the technical discussion (pull-request).
+
+This helps us to reach several goals:
+* Maintain the high quality standards of Citrus
+* Make changes transparent to the community
+* Discuss changes to ensure that the best possible solution will make it into the next release
+* Enable a sustainable system for Citrus maintainers to review contributions
+
+>**Important Note**: If you'd like to propose a pull request for a bug fix, please ensure that your base branch is
+`v2.7-bugfix` instead of master and that the target branch for the pull request is `v2.7-bugfix` as well .
+
+The pull request workflow is as follows: 
+* Every pull request will be built via [Travis CI](https://travis-ci.org/citrusframework/citrus). It's mandatory that
+  the build of the pull request is successful before the review begins. 
+* A maintainer will [review](#review-criteria) your changes and provide feedback to you.
+* If the pull request is project internal, it is required that the 
+  [quality gate](https://sonarcloud.io/dashboard?id=citrus) is fulfilled.
+  
+#### Review criteria
+The review of proposed code will focus on some important criteria for this project.
+
+* The code should follow common style and design principles as well as our [sonarcloud rule set](https://sonarcloud.io/dashboard?id=citrus)
+* The code should have a suitable test coverage
+* The changes should be documented
+
+### Types of change
+We distinguish between different types of change which indicate different level of complexity, scope and required
+knowledge about the Citrus framework.
+ 
+* **Maintenance**  
+  Tasks required to keep the framework up to date. E.g. updating dependencies. Depending on the tasks, 
+  the changes may vary from smaller to larger size with a according time required. Most of the tasks just require a
```

Comment:
(Depending on the tasks, the changes may vary from smaller to larger size with a according time required)
Tasks may vary in size from smaller to larger changes.